09-Photutils/photutils_overview_solutions.ipynb

```diff
@@ -0,0 +1,130 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Photutils Overview"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise solutions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`data/xdf_hst_wfc3ir_60mas_f105w_sci.fits` is an F105W image of the same field used for the preceding examples. These images are aligned, so source positions detected in the F160W image should correspond to the same source if they are visible in the F105W image. \n",
+    "\n",
+    "In the F105W image, find the countrate in a 5-pixel aperture radius of the 3 brightest sources detected in the F160W image (hint: read in the table of detected sources that was written to disk in the previous example)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy.io import fits\n",
+    "from astropy.table import Table\n",
+    "from photutils import CircularAperture, aperture_photometry"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# read in the data \n",
+    "f105w_sci_hdulist = fits.open('data/xdf_hst_wfc3ir_60mas_f105w_sci.fits')\n",
+    "f105w_sci_data = f105w_sci_hdulist[0].data\n",
+    "\n",
+    "# read in the table of sources detected in the F160W image\n",
+    "detected_sources = Table.read('my_detected_sources.txt', format='ascii.ecsv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# sort table by source_sum column and select the 3 brightest\n",
+    "detected_sources.sort('source_sum')\n",
+    "detected_sources.reverse()\n",
+    "three_brightest = detected_sources[0:3]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pull out x, y coordinates from table\n",
+    "x = three_brightest['xcentroid']\n",
+    "y = three_brightest['ycentroid']\n",
+    "\n",
+    "# make list of tuples of coordinate pairs\n",
+    "positions = list(zip(x, y))\n",
+    "positions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# construct 5-pixel radius circular aperture object\n",
+    "aperture = CircularAperture(positions, r=5)\n",
+    "\n",
+    "# run aperture photometry\n",
+    "phot = aperture_photometry(f105w_sci_data, aperture)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# inspect photometry table. All sources are brighter in F105W than they are in F160W\n",
```

Comment:
Actually, this isn't true because the photometry was measured in different "apertures":  one in a circular aperture and the other using the isophotal segment.  To do a fair comparison, one would need to also do aperture photometry (identical aperture) on the F160W image:

```python
>>> f160w_sci_hdulist = fits.open('data/xdf_hst_wfc3ir_60mas_f160w_sci.fits')
>>> f160w_sci_data = f160w_sci_hdulist[0].data
>>> phot_f160w = aperture_photometry(f160w_sci_data, aperture)
>>> phot['aperture_sum_f160w'] = phot_f160w['aperture_sum']
>>> phot
 id      xcenter            ycenter          aperture_sum    aperture_sum_f160w
           pix                pix                                              
--- ------------------ ------------------ ------------------ ------------------
  1 186.94149524009447 119.33278672140875 1.9365915934381521  1.468894044498522
  2  159.1530544625218  62.67425991294218 0.8595147855407592 0.7192989780268467
  3  73.60684792073822 139.45544954074361 0.5643753267720801 0.6152713356202307
```

In a r=5 pix aperture, source `id=3` is brighter in F160W than F105W.

It may be good to add this to the Exercise (ask them to compare the aperture photometry in the F105W and F160W data; which source is brighter in F160W?).

Reply:
Maybe as a "Bonus" for those that finish early?